### PR TITLE
Add Terminology, APIs, Behaviour sections and ref to IS-04 versioning mechanism

### LINKED ISSUE #7: Changes in this API MUST trigger IS-04 version changes
which can be followed through the notification mechanism of a registry.

### PR DESCRIPTION
**Pull request by garethsb**
Resolves #7

### CODE REVIEW

**Review by peterbrightwell — APPROVED** [review]
Looks good

**Comment by garethsb** [comment]
Also now addresses  #3, #4 and #6.

File: docs/Behaviour.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+The reset behaviour depends on the API implementation.
+The new values MUST be valid per the schema.
+
+* For labels and descriptions, the implementation MAY restore an initial or configured default value, or set the value to the empty string.
+* For a named tag, the implementation MAY restore an initial or configured default array of values, or remove the named tag from the resource.
+
+A `PATCH` request with `tags` set to `null` asks to reset all tags.
+
+The following `PATCH` request asks to completely reset the distinguishing information for a resource.
```
Comment (cristian-recoseanu) [inline docs/Behaviour.md]:
```suggestion
The following `PATCH` request asks to completely reset the label, description and tags for a resource.
```

**Comment by peterbrightwell** [comment]
Call 2023-05-18: 

- Change "Distinguishing Information" to prevent confusion with BCP-002-02
- Consider correctness of "User", and if glossary needs improving
- Updates MUST be persistent (not just SHOULD)

@garethsb will make these changes now and we will merge then make any tweaks on later PRs/v1.0-dev

**Comment by garethsb** [comment]
"Distinguishing information" --> "core information properties"

**Pull request merged**